python/ray/data/dataset.py

```diff
@@ -5245,20 +5245,14 @@ def to_dask(
             pa = None
 
         from ray.data._internal.pandas_block import PandasBlockSchema
-        from ray.util.client.common import ClientObjectRef
         from ray.util.dask import ray_dask_get
 
         dask.config.set(scheduler=ray_dask_get)
 
         @dask.delayed
         def block_to_df(block_ref: ObjectRef[Block]) -> pd.DataFrame:
-            if isinstance(block_ref, (ray.ObjectRef, ClientObjectRef)):
-                raise ValueError(
-                    "Dataset.to_dask() must be used with Dask-on-Ray, please "
-                    "set the Dask scheduler to ray_dask_get (located in "
-                    "ray.util.dask)."
-                )
-            return _block_to_df(block_ref)
+            block = ray.get(block_ref)
+            return _block_to_df(block)
```

Comment:
Let me double-check the right behavior.

Reply:
Reworked in https://github.com/ray-project/ray/pull/52589/commits/ed174252a5045662a6ccb55a757f8e6ce657f7b8